aiocache/decorators.py

```diff
@@ -328,7 +328,7 @@ async def decorator(
         result = await f(*new_args, **kwargs)
         result.update(partial)
 
-        if cache_write:
+        if result and cache_write:
```

Comment:
As a user... with this (not that before it worked as expected) it means that the execution will be executed every time the results are empty, is that the behavior you would expect?

Reply:
Good point, in my case I already had something like:

```py
if not params:
    return {}
```

So it wouldn't make any difference. But I can understand that in some cases you might want to do some logic even if the params are empty because everything is cached already.

I would be against putting a new configuration to control this (too many already), so there are 2 options:

1. Always call the method even if the params are empty. If you don't want to do anything a shortcut can be added at the beginning of the method to just return, so the behaviour can be controlled.
2. Never call the method when the params are empty. In this case if I wanted to do something in the method (like logging) I would not have any workaround.

I would go for option 1 then.

Reply:
this looks similar to https://github.com/argaen/aiocache/issues/452. Maybe the 3rd option is what @paulo-raca mentions in the issue of having an "empty" token and write this into Redis (and others). Wdyt?

Reply:
Yes, that would work too, since in that case we would be writing the "guard" value internally instead of the empty dict that causes the error.

Reply:
I think that discussion relates more to the other decorators, where the expected return value may be `None`.

For multi_cached though, we should clearly define what an empty dict signifies. I suspect this change is fine and it can be treated as a 'no-cache' for the result, indicating that something failed temporarily and a result couldn't be retrieved. Alternatively, it could simply be an error and a function is expected to always return values for the keys requested. Or it could be treated as a normal result that should be cached.

I'm not sure what the use cases are for this though.